app/src/main/kotlin/com/intive/picover/profile/viewmodel/ProfileViewModel.kt

```diff
@@ -25,14 +27,10 @@ class ProfileViewModel @Inject constructor(
 	@Validator.ShortText private val textValidator: TextValidator,
 ) : StatefulViewModel<Profile>() {
 
-	val userName = derivedStateOf {
-		_state.value.let { state ->
-			if (state is Loaded) {
-				state.data.name
-			} else {
-				""
-			}
-		}
+	private val _username = mutableStateOf("")
+	val username: State<String> = _username
+	val usernameErrorMessageId = derivedStateOf {
```

Comment:
Do we actually need `derivedStateOf` here, if each change on `username` triggers recomposition anyway?

Reply:
Done. There was an unnecessary recomposition trigger. Thanks for catching it 🍻